inc/taskjobview.class.php

```diff
@@ -462,13 +463,33 @@ public function ajaxModuleItemsDropdown($options)
                     'agents.id',
                     'agents.items_id'
                 ]
-            ]);
+            ];
+
+            $item = getItemForItemtype($itemtype);
+            if ($item->isEntityAssign()) {
+                // get entity ID from task ID
+                $pfTask = new PluginGlpiinventoryTask();
+                $pfTask->getFromDB($taskid);
+
+                $crit['WHERE'] += getEntitiesRestrictCriteria(
+                    'agents',
+                    '',
+                    $pfTask->fields['entities_id'] ?? 0,
+                    $item->maybeRecursive()
+                );
+            }
+
+            $iterator = $DB->request($crit);
+
```

Comment:
An option exist to no limited task to current entity (for `networkinventory)` see

![image](https://github.com/user-attachments/assets/7c939c43-4e4e-466b-b7f0-f00f7409219b)

Reply:
The checkbox "Restrict scope to task entity" doesn't have the same intention than this portion of the code. The "Restrict scope to task entity" applies only for IP ranges (Target dropdown). This PR applies for Agent dropdown to gather only Agents assigned to the same Entity of the Task.

Reply:
indeed :+1: